### PR TITLE
[GPU] Add `use_onednn_impls` layout optimizer attribute

### PR DESCRIPTION
**Pull request by sshlyapn**
This PR adds new attribute `use_onednn_impls` to layout_optimizaer

### CODE REVIEW

**Review by lznamens — APPROVED** [review]

**Review by isanghao — APPROVED** [review]

File: inference-engine/thirdparty/clDNN/src/program.cpp [hunk 1]
```diff
@@ -1374,4 +1374,9 @@ void program::set_layout_optimizer_attributes(layout_optimizer& lo) {
 
     if (should_use_bs_fs_yx_bsv16_fsv16)
         lo.set_optimization_attribute(layout_optimizer::optimization_attributes_type::bs_fs_yx_bsv16_fsv16_network, 1);
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    if (dev_info.supports_immad)
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/program.cpp:1379]: I realized that we need to check both `supports_immad` and `queue_type == in_order`. Otherwise, layout optimizer will try to use onednn primitive for unit tests.

File: inference-engine/thirdparty/clDNN/src/program.cpp [hunk 2]
```diff
@@ -1374,4 +1374,9 @@ void program::set_layout_optimizer_attributes(layout_optimizer& lo) {
...
+        lo.set_optimization_attribute(layout_optimizer::optimization_attributes_type::use_onednn_impls, 1);
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/program.cpp:1380]: I realized that we need to check both supports_immad and queue_type == in_order. Otherwise, layout optimizer will try to use onednn primitive for unit tests.

**Review by isanghao — CHANGES_REQUESTED** [review]

File: inference-engine/thirdparty/clDNN/src/program.cpp (hunk 2 above)
Comment (sshlyapn) [reply inference-engine/thirdparty/clDNN/src/program.cpp:1380]: Thanks, added

File: inference-engine/thirdparty/clDNN/src/program.cpp [hunk 3]
```diff
@@ -1374,4 +1374,9 @@ void program::set_layout_optimizer_attributes(layout_optimizer& lo) {
 
     if (should_use_bs_fs_yx_bsv16_fsv16)
         lo.set_optimization_attribute(layout_optimizer::optimization_attributes_type::bs_fs_yx_bsv16_fsv16_network, 1);
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/program.cpp:1378]: just out of curiosity: why did you change the spec of `set_layout_optimizer_attributes`? I think it is possible to call `get_engine` within the function.

File: inference-engine/thirdparty/clDNN/src/program.cpp [hunk 4]
```diff
@@ -1374,4 +1374,9 @@ void program::set_layout_optimizer_attributes(layout_optimizer& lo) {
...
+    if (engine.get_device_info().supports_immad && engine.configuration().queue_type == queue_types::in_order)
+        lo.set_optimization_attribute(layout_optimizer::optimization_attributes_type::use_onednn_impls, 1);
+#endif
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/program.cpp:1381]: nit: I think it will cause unused variable warning here if `!ENABLE_ONEDNN_FOR_GPU`

File: inference-engine/thirdparty/clDNN/src/program.cpp (hunk 3 above)
Comment (sshlyapn) [reply inference-engine/thirdparty/clDNN/src/program.cpp:1378]: You are absolutely right! My fault

**Review by isanghao — APPROVED** [review]
